### PR TITLE
Extracted language strings

### PR DESCRIPTION
**Pull request by xRealNeon**
Part 1 of issue: #616

### CODE REVIEW

**Comment by dyc3** [comment]
Awesome work! The tests seem to be failing though. I've seen the error before, you have to add the `i18n` instance to each of the places where the component is mounted using `mount` or `shallowMount`.

Example:

https://github.com/dyc3/opentogethertube/blob/a32975b60ecfde3fbc33dc668a841a7ec007d213/client/tests/unit/Chat.spec.js#L25-L32

File: client/src/views/Privacy.vue [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
 
     <p>
-      If you have registered an account, your email is only used for account recovery, or to contact you in case it is necessary to resolve a bug, or follow up on site feedback. An email is not required if you log in with Discord. Your email, owned rooms, or other account information is your private information, and as such will never be distributed or sold to third parties.
+      {{ $t("privacy.text4") }}
     </p>
 
     <p>
-      OpenTogetherTube is GDPR compliant because of how little information is collected. If, for some reason, you really want to have the one database row that represents your account, reach out to me <a href="https://twitter.com/rollthedyc3">on twitter.</a>
+      {{ $t("privacy.text5.text") }}
+      <a href="https://twitter.com/rollthedyc3">{{ $t("privacy.text5.link-text") }}</a>
```
[the remark this answers is not in the document]
Comment (xRealNeon) [reply client/src/views/Privacy.vue:22]: You are right, maybe add an empty string behind the link for the grammar?

**Comment by dyc3** [comment]
Hmm, that's a weird one. You wanna commit what you have so far so I can reproduce it?

**Comment by dyc3** [comment]
Ah, for components that are in typescript, it appears that we are supposed to use the `useI18n` function to import the `t()` function, and that will give us the correct typing. Although the examples use `defineComponent`, I'm pretty sure you can get it to work using the class components we have.

```ts
const { t, d, n } = useI18n({
      inheritLocale: true
})
```

**Comment by xRealNeon** [comment]
What do I have to do where?

**Comment by dyc3** [comment]
For all the components that are giving you errors, use `t()` from:

```ts
import { useI18n } from 'vue-i18n'

const { t, d, n } = useI18n({
      inheritLocale: true
})
```

instead of `this.$t()`.

The reason we are getting these errors is because typescript can't magically know the type of `this.$t`, because it doesn't exist in the type declarations.

**Comment by xRealNeon** [comment]
Ok ill do it tomorrow, it's 23:14 😫

**Comment by dyc3** [comment]
Ah, wait a second, we have to update `vue-i18n` in order to use that. Whoops! I'll try to update that for you.

**Comment by dyc3** [comment]
Turns out we are going to need to upgrade to vue3 to fix this properly. However, I _really_ wanna get this merged, so for now use this to ignore all the lines where its throwing type errors:

```ts
// @ts-expect-error Ignore type checking for i18n, pending Vue 3 upgrade
```

**Comment by xRealNeon** [comment]
Mhh maybe I don't understand, or why isn't it working?

grafik
grafik

**Pull request merged**